pkg/console/subresource/configmap/configmap_test.go

```diff
@@ -105,14 +106,30 @@ func TestDefaultConfigMap(t *testing.T) {
 					Finalizers:      nil,
 					ClusterName:     "",
 				},
-				Data:       map[string]string{"console-config.yaml": exampleYaml},
+				Data:       map[string]string{configKey: exampleYaml},
 				BinaryData: nil,
 			},
 		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			if diff := deep.Equal(DefaultConfigMap(tt.args.operatorConfig, tt.args.consoleConfig, tt.args.infrastructureConfig, tt.args.rt), tt.want); diff != nil {
+			cm, _, _, _ := DefaultConfigMap(tt.args.operatorConfig, tt.args.consoleConfig, tt.args.infrastructureConfig, tt.args.rt)
+			// Deep.Equal doesn't work well for checking the value of the embedded
+			// console-config.yaml file, as it is just a string.  The MergeProcessConfig()
+			// call within DefaultConfigMap() converts the value to JSON to do the merge,
+			// then converts back, therefore we can't trust the ordering of the strings.
+			//
+			// so here we are going to just check that we have a value
```

Comment:
Why not parse the string as YAML and make sure the result matches what's expected?

Reply:
Well, if we are just doing a `string(SomeYAML)` here https://github.com/openshift/console-operator/pull/162/files#diff-ffa541898964e88404798d9f8c923256R121

Then the `TestNewYamlConfig()` test is really already doing the testing we need on this bit.

I imagine if I were going to parse it to check, I'd turn it back into a `go struct` and do more or less the same thing again.

Reply:
Considering if i should parse it into [the actual struct](https://github.com/openshift/console/blob/master/cmd/bridge/config.go#L13):

```golang
// https://github.com/openshift/console/blob/master/cmd/bridge/config.go#L13
// Config is the top-level console configuration.
type Config struct {
	APIVersion    string `yaml:"apiVersion"`
	Kind          string `yaml:"kind"`
	ServingInfo   `yaml:"servingInfo"`
	ClusterInfo   `yaml:"clusterInfo"`
	Auth          `yaml:"auth"`
	Customization `yaml:"customization"`
}

// ServingInfo holds configuration for serving HTTP.
type ServingInfo struct {
	BindAddress string `yaml:"bindAddress"`
	CertFile    string `yaml:"certFile"`
	KeyFile     string `yaml:"keyFile"`

	// These fields are defined in `HTTPServingInfo`, but are not supported for console. Fail if any are specified.
	// https://github.com/openshift/api/blob/0cb4131a7636e1ada6b2769edc9118f0fe6844c8/config/v1/types.go#L7-L38
	BindNetwork           string        `yaml:"bindNetwork"`
	ClientCA              string        `yaml:"clientCA"`
	NamedCertificates     []interface{} `yaml:"namedCertificates"`
	MinTLSVersion         string        `yaml:"minTLSVersion"`
	CipherSuites          []string      `yaml:"cipherSuites"`
	MaxRequestsInFlight   int64         `yaml:"maxRequestsInFlight"`
	RequestTimeoutSeconds int64         `yaml:"requestTimeoutSeconds"`
}

// ClusterInfo holds information the about the cluster such as master public URL and console public URL.
type ClusterInfo struct {
	ConsoleBaseAddress string `yaml:"consoleBaseAddress"`
	ConsoleBasePath    string `yaml:"consoleBasePath"`
	MasterPublicURL    string `yaml:"masterPublicURL"`
}

// Auth holds configuration for authenticating with OpenShift. The auth method is assumed to be "openshift".
type Auth struct {
	ClientID            string `yaml:"clientID"`
	ClientSecretFile    string `yaml:"clientSecretFile"`
	OAuthEndpointCAFile string `yaml:"oauthEndpointCAFile"`
	LogoutRedirect      string `yaml:"logoutRedirect"`
}

// Customization holds configuration such as what logo to use.
type Customization struct {
	Branding             string `yaml:"branding"`
	DocumentationBaseURL string `yaml:"documentationBaseURL"`
}
```
and compare the objects then.  But that would require pulling the `type Config` from `Console` repo.

Reply:
Nope. Solved another way.